python-threatexchange/threatexchange/cli/fetch_cmd.py

```diff
@@ -195,6 +195,17 @@ def execute_for_collab(
                 next_checkpoint = delta.checkpoint
                 self._fetch_progress(len(delta.updates), next_checkpoint)
                 assert next_checkpoint is not None  # Infinite loop protection
+                if checkpoint is not None:
+                    prev_time = checkpoint.get_progress_timestamp()
+                    next_time = next_checkpoint.get_progress_timestamp()
+                    if prev_time and next_time:
+                        assert prev_time <= next_time, (
```

Comment:
blocking: HMA will probably retry `execute_for_collab` here. Can we throw a custom exception so that we do a finite number of retries (or none at all)?

HTTP errors are common and we probably want a logic like: Keep retrying this operation unless it is a certain class of problems.

If we throw a custom exception, we can enumerate the 'non-retryable' problems.

Reply:
Now that I read this a bit more, it seems like HMA will need to re-implement the entirety of fetch_cmd. Proposal, I am writing the HMA side as we speak. 

How about I implement a FetchRunner class in pytx that can replace large parts of fetch_cmd AND work for HMA?

Reply:
Let's sync offline, I think defining custom exceptions might involve interface changes because exceptions might be thrown in fetch itself.